### DISCUSSION TITLE
v3: Filtering a list with nested components crashes

### DISCUSSION THREAD

**Discussion by robindrost** [livewire/livewire#5802] [7. Bugs (archived)]
Hello,

I just upgraded to v3 and have an issue on a list with a search field and sub components. When I search and it matches one of the items in the list the sub component seem to crash the list.

**The error:**

```
Uncaught TypeError: "Snapshot missing on Livewire component with id: " is not a constructor
    at new Component (livewire.js?id=3605227a:3798:15)
    at initComponent (livewire.js?id=3605227a:3846:21)
    at livewire.js?id=3605227a:7848:26
    at livewire.js?id=3605227a:1591:58
    at livewire.js?id=3605227a:1189:42
    at Array.forEach (<anonymous>)
    at livewire.js?id=3605227a:1189:27
    at walk (livewire.js?id=3605227a:1112:5)
    at livewire.js?id=3605227a:1187:7
    at deferHandlingDirectives (livewire.js?id=3605227a:1002:5)
```

Here is an example of the code:

**UserTest (the main page component)**
```
<?php

namespace App\Livewire;

use App\Models\User;
use Livewire\Component;

class UserTest extends Component
{
    public $search;

    public function render()
    {
        return view('livewire.user-test', [
            'users' => User::when($this->search, function ($query) {
                $query->where('name', 'like', '%'.$this->search.'%');
            })->get(),
        ]);
    }
}
```

**The view:**

```
<div>
    <div>
        <input wire:model.live="search" />
    </div>

    @foreach ($users as $user)
        <div class="flex">
            <span>{{ $user->name }}</span>
            <livewire:test-sub :key="$user->id" />
        </div>
    @endforeach
</div>
```

**And the sub component that is just a string:**
```
<?php

namespace App\Livewire;

use Livewire\Component;

class TestSub extends Component
{
    public function render()
    {
        return view('livewire.test-sub');
    }
}
```

```
<div>
    Test
</div>
```

Am I missing something that is new in v3? I have read the documentation on nested components but cannot find anything.

Kinds regards

**Comment by nuernbergerA** [comment]
i think this should work
```blade
<livewire:test-sub :wire:key="$user->id" />
// or
<livewire:test-sub wire:key="{{ $user->id }}" />
```

**Reply by robindrost** [reply]
Thanks for the reply. Unfortunately this does not fix the problem.

**Comment by PhiloNL** [comment]
I think it is related to: https://github.com/livewire/livewire/pull/5776
It should be fixed once https://github.com/alpinejs/alpine/pull/3658 is merged.

**Reply by robindrost** [reply]
Thanks @PhiloNL! I'll check back when the pr is merged and see if the problem is resolved :)!

**Comment by robindrost** [comment]
~~So it turns out you have to apply a key to every child element, not only the child livewire component, of the foreach to make it work 😄.~~

Fixed in an update of LiveWire!

**Reply by joshhanley** [reply]
@robindrost hmm that shouldn't be the case, so it's probably a bug. Can you share what your components look like now (with all the keys)?

**Reply by robindrost** [reply]
After trying to make it fail again it still works. Might have been fixed in an update with Livewire in the past days :)!
